lib/bolt/transport/ssh/connection.rb

```diff
@@ -68,6 +68,8 @@ def initialize(target, transport_logger, load_config = true)
           require 'net/ssh'
           require 'net/ssh/proxy/jump'
 
+          raise Bolt::ValidationError, "Target #{target.name} does not have a host" unless target.host
+
```

Comment:
Phew ok finally figured it out - we need to add `host` to the options hash of each transport https://github.com/puppetlabs/bolt/blob/master/lib/bolt/transport/ssh.rb#L12